packages/tempus/unit_test/Tempus_UnitTest_BackwardEuler.cpp

```diff
@@ -79,25 +83,346 @@ TEUCHOS_UNIT_TEST(BackwardEuler, Construction)
+// ************************************************************
+class StepperBackwardEulerModifierTest
+  : virtual public Tempus::StepperBackwardEulerModifierBase<double>
+{
+public:
+
+  /// Constructor
+  StepperBackwardEulerModifierTest()
+    : testBEGIN_STEP(false), testBEFORE_SOLVE(false),
+      testAFTER_SOLVE(false), testEND_STEP(false),
+      testCurrentValue(-0.99), testWorkingValue(-0.99),
+      testDt(-1.5), testType("")
+  {}
+
+  /// Destructor
+  virtual ~StepperBackwardEulerModifierTest(){}
+
+  /// Modify BackwardEuler Stepper at action location.
+  virtual void modify(
+    Teuchos::RCP<Tempus::SolutionHistory<double> > sh,
+    Teuchos::RCP<Tempus::StepperBackwardEuler<double> > stepper,
+    const typename Tempus::StepperBackwardEulerAppAction<double>::ACTION_LOCATION actLoc)
+  {
+    switch(actLoc) {
+      case StepperBackwardEulerAppAction<double>::BEGIN_STEP:
+      {
+        testBEGIN_STEP = true;
+        auto x = sh->getCurrentState()->getX();
+        testCurrentValue = get_ele(*(x), 0);
+        break;
+      }
+      case StepperBackwardEulerAppAction<double>::BEFORE_SOLVE:
+      {
+        testBEFORE_SOLVE = true;
+        testDt = sh->getWorkingState()->getTimeStep()/10.0;
+        sh->getWorkingState()->setTimeStep(testDt);
+        break;
+      }
+      case StepperBackwardEulerAppAction<double>::AFTER_SOLVE:
+      {
+        testAFTER_SOLVE = true;
+        testType = "Backward Euler - Modifier";
+        stepper->setStepperType(testType);
+        break;
+      }
+      case StepperBackwardEulerAppAction<double>::END_STEP:
+      {
+        testEND_STEP = true;
+        auto x = sh->getWorkingState()->getX();
+        testWorkingValue = get_ele(*(x), 0);
+        break;
+      }
+      default:
+        TEUCHOS_TEST_FOR_EXCEPTION(true, std::logic_error,
+        "Error - unknown action location.\n");
+    }
+  }
+
+  bool testBEGIN_STEP;
+  bool testBEFORE_SOLVE;
+  bool testAFTER_SOLVE;
+  bool testEND_STEP;
+  double testCurrentValue;
+  double testWorkingValue;
+  double testDt;
+  std::string testType;
+};
+
+TEUCHOS_UNIT_TEST(BackwardEuler, AppAction_Modifier)
+{
+  Teuchos::RCP<const Thyra::ModelEvaluator<double> >
+    model = rcp(new Tempus_Test::SinCosModel<double>());
+
+  // Setup Stepper for field solve ----------------------------
+  auto stepper = rcp(new Tempus::StepperBackwardEuler<double>());
+  stepper->setModel(model);
+  auto modifier = rcp(new StepperBackwardEulerModifierTest());
+  stepper->setAppAction(modifier);
+  stepper->initialize();
+
+  // Create a SolutionHistory.
+  auto solutionHistory = Tempus::createSolutionHistoryME(model);
+
+  // Take one time step.
+  stepper->setInitialConditions(solutionHistory);
+  solutionHistory->initWorkingState();
+  double dt = 0.1;
+  solutionHistory->getWorkingState()->setTimeStep(dt);
+  stepper->takeStep(solutionHistory);
+
+  TEST_COMPARE(modifier->testBEGIN_STEP, ==, true);
```

Comment:
@ccober6 Could you add some comments on what these tests on the modifier are checking? Is it that each of the ACTION_LOCATION's have been called?

Reply:
@krcb Will do. Yes, just checking that each ACTION_LOCATION has been called.  I have also done this for the Explicit RK testing.

Reply:
@ccober6 Ok, this makes sense. Based on [this comment](https://github.com/trilinos/Trilinos/pull/6808#discussion_r406841146), it's not possible to perform a runtime test of what happens if the wrong action is passed in, as this will result in a compile time error?